packages/client/components/TeamPrompt/WorkDrawer/LinearIntegrationResults.tsx

```diff
@@ -0,0 +1,164 @@
+import useLoadNextOnScrollBottom from '../../../hooks/useLoadNextOnScrollBottom'
+import Ellipsis from '../../Ellipsis/Ellipsis'
+import LinearObjectCard from './LinearObjectCard'
+
+type LinearIssueEdge = NonNullable<
+  NonNullable<
+    NonNullable<
+      NonNullable<
+        LinearIntegrationResults_query$data['viewer']['teamMember']
+      >['integrations']['linear']['api']
+    >['query']
+  >['issues']['edges']
+>[number]
+
+// Placeholder type for the issue node, replace with generated type later
+type LinearIssueNode = NonNullable<LinearIssueEdge>['node']
+
+interface Props {
+  queryRef: PreloadedQuery<LinearIntegrationResultsQuery>
+  teamId: string
+}
+
+const LinearIntegrationResults = (props: Props) => {
+  const {queryRef, teamId} = props
+
+  // This query definition must match the one used to generate the queryRef in the parent component (LinearIntegrationResultsRoot)
+  // Assuming the filter variable passed down is a string intended for title search.
+  const queryData = usePreloadedQuery<LinearIntegrationResultsQuery>(
+    graphql`
+      query LinearIntegrationResultsQuery($teamId: ID!, $filter: _xLinearIssueFilter) {
+        # Pass the filter string to the fragment
+        ...LinearIntegrationResults_query @arguments(teamId: $teamId, filter: $filter)
+      }
+    `,
+    queryRef
+  )
+
+  // This fragment defines the data structure and pagination logic for Linear issues
+  const paginationFragment = graphql`
+    fragment LinearIntegrationResults_query on Query
+    @argumentDefinitions(
+      cursor: {type: "String"}
+      count: {type: "Int", defaultValue: 25}
+      teamId: {type: "ID!"}
+      filter: {type: "_xLinearIssueFilter"}
+    )
+    @refetchable(queryName: "LinearIntegrationResultsPaginationQuery") {
+      # The filter is applied directly to the 'issues' connection below
+      viewer {
+        teamMember(teamId: $teamId) {
+          integrations {
+            linear {
+              api {
+                errors {
+                  message
+                  locations {
+                    line
+                    column
+                  }
+                  path
+                }
+                query {
+                  issues(
+                    first: $count
+                    after: $cursor
+                    filter: $filter
+                    sort: {updatedAt: {order: Descending}}
+                  ) @connection(key: "LinearIntegrationResults_query_issues") {
+                    edges {
+                      node {
+                        ... on _xLinearIssue {
+                          id
+                          identifier
+                          title
+                          url
+                          project {
+                            name
+                          }
+                          team {
+                            displayName
+                          }
+                          # Spread the fragment for LinearObjectCard
```

Comment:
-1 reading these AI comments makes me mad. I think it was trained on too many tutorials and examples.